### DISCUSSION TITLE
Temporary Containers

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#821] [Ideas]
### What feature would you like to see?

It would be create to have a similar functionality that Temporary Containers (https://github.com/stoically/temporary-containers) offers.
It is possible to open a new tab/window in a new temporary container, no cookies and no history everything from zero, every time - meaning also the tracking is limited. This will obviously not counter finger printing but its a step into the right direction.

### What platform are you seeing the problem on?

macOS

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/487_

**Comment by AverageHelper** [comment]
Currently, it seems the plugin's Automatic Mode does not properly detect the opening of new tabs in Zen Browser. If that could be fixed, or if Zen would include the plugin's functionality like it does [Firefox Multi-Account Containers](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/), that'd be wonderful!

**Comment by RealDishwash** [comment]
What's the benefit of this feature compared to opening in incognito mode?

**Reply by AverageHelper** [reply]
More control over which tabs share cookies than just two window types. Also, only cookies get silo'd by default, history is retained.

**Reply by me-nkr** [reply]
also isolated temporary windows, so it can be used for development and testing

**Reply by d5364t54ytfr4** [reply]
These temporary containers behave exactly like normal containers, except they are deleted every x minutes.
Can be used for bypassing `Private Mode` detection. For example `News paper` websites that only give a few articles to read.

So basically:
- To avoid private mode detection in websites and other flaws of private mode. Because Temporary containers work exactly like normal container tabs.
- I think the history is also saved, but not the cookies and other site data.
- To have private tabs in the same window instead of in another window.

**Comment by vraravam** [comment]
I too would love this feature. The way I use FF with both Multi-Account Containers and Temporary Containers is that whenever I open a new tab, it automatically opens in a temporary container. I can then choose to reopen the same in one of the named containers. That choice is vital.
Currently, in Zen, I have configured MAC and TempContainers, but in the workspaces, if I associate a container to a workspace, any link I open in that workspace - it opens a new tab in that default container only. I would like for this last part to not be the case. (I have tried with resetting the default container mapped to the workspace as well - even in this case, any link opened from a named container (for eg Personal) opens a new tab only in a Personal container.

**Comment by d5364t54ytfr4** [comment]
Floorp Browser(another Firefox based browser) has a Private container option. You can right click on any link and you get the `Open in  New Tab with Private Container` option.

Btw `Container Sidebar` is really useful in normal Firefox.

Here is a duplicate of the issue:
https://github.com/zen-browser/desktop/discussions/767
And here's a compliation of feature requests that mentions this:
https://github.com/zen-browser/desktop/discussions/2078
